### PR TITLE
[IMP]10728 add module delivery_carrier_extend.

### PR DESCRIPTION
**Pull request by Rona111**

### CODE REVIEW

**Comment by elicoidal** [comment]
Name to be improved
Headers/README/PEP8

**Comment by Rona111** [comment]
cc @elicoidal @duanyp1991 @seb-elico @siyuancn 
 I modified the module's name and fix the RADME/openerp.py/PEP8,also tested it on the v8.

**Comment by elicoidal** [comment]
name is too complex
delivery_carrier_percentage_on_price

File: delivery_methods_based_on_percentage_of_sale_price/delivery_extend.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+class delivery_carrier(osv.osv):
+    '''
+    Add new filed to record the percentage of the total price if  pay by paypal.
+    '''
+    _inherit = "delivery.carrier"
+    _name = "delivery.carrier"
+
+    _columns = {
+        'percentage': fields.float('The Percentage  of price',
```
Comment (elicoidal) [inline delivery_methods_based_on_percentage_of_sale_price/delivery_extend.py]: Percentage of Sales price

File: delivery_methods_based_on_percentage_of_sale_price/delivery_extend.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                                   required=True,
+                                   help='The percentage of the total price.\
```
Comment (elicoidal) [inline delivery_methods_based_on_percentage_of_sale_price/delivery_extend.py]: Percentage of delivery based compared with sales price

**Comment by elicoidal** [comment]
some details. other than that, LGTM

**Comment by Rona111** [comment]
cc @elicoidal @siyuancn @duanyp1991 @seb-elico 
finished the name modified.

File: delivery_carrier_percentage_on_price/delivery_extend_view.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <record id="delivery_form_view_extend" model="ir.ui.view">
```
Comment (siyuancn) [inline delivery_carrier_percentage_on_price/delivery_extend_view.xml]: view name: view_delivery_carrier_form

**Comment by elicoidal** [comment]
LGTM

**Comment by Rona111** [comment]
cc @elicoidal @siyuancn @duanyp1991 @seb-elico

File: delivery_carrier_percentage_on_price/delivery_extend.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    '''
+    Add new filed to record the percentage of the total price if pay by paypal.
+    '''
+    _inherit = "delivery.carrier"
+    _name = "delivery.carrier"
+
+    _columns = {
+        'percentage': fields.float('Percentage of Sales price',
+                                   required=True,
+                                   help='Percentage of delivery based compared with sales price.\
```
Comment (elicoidal) [inline delivery_carrier_percentage_on_price/delivery_extend.py:24]: Percentage of delivery compared with sales price

File: delivery_carrier_percentage_on_price/delivery_extend.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                                   Value between 0~100.'),
```
Comment (elicoidal) [inline delivery_carrier_percentage_on_price/delivery_extend.py:25]: you should add an SQL constraint to avoid wrong values

**Pull request merged**